src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -325,6 +330,96 @@ create_reference_table(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * create_citus_local_table creates a citus table from the table with relationId.
+ * The created table would have the following properties:
+ *  - it will have only one shard,
+ *  - its distribution method will be DISTRIBUTE_BY_NONE,
+ *  - its replication model will be ReplicationModel,
+ *  - its replication factor will be set to 1.
+ * On the contrary of reference tables, a citus local table has only one placement
+ * and that placement will only be in the coordinator for now.
+ * In fact, the above is the definition of a citus local tables.
+ */
+Datum
+create_citus_local_table(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	Oid relationOid = PG_GETARG_OID(0);
+
+	/*
+	 * Lock target relation with an exclusive lock - there's no way to make
+	 * sense of this table until we've committed, and we don't want multiple
+	 * backends manipulating this relation.
+	 */
+	Relation relation = try_relation_open(relationOid, ExclusiveLock);
+
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("could not create citus local table: "
+							   "relation does not exist")));
+	}
+
+	ErrorIfUnsupportedCreateCitusLocalTable(relationOid);
+
+	ObjectAddress tableAddress = { 0 };
+	ObjectAddressSet(tableAddress, RelationRelationId, relationOid);
+
+	/*
+	 * Ensure dependencies first as we will create shell table on the other nodes
+	 * in the MX case.
+	 */
+	EnsureDependenciesExistOnAllNodes(&tableAddress);
+
+	CreateCitusLocalTable(relationOid);
+
+	relation_close(relation, NoLock);
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * ErrorIfUnsupportedCreateCitusLocalTable errors out if we cannot create the
+ * citus local table from the relation relationOid.
+ */
+static void
+ErrorIfUnsupportedCreateCitusLocalTable(Oid relationOid)
+{
+	/* citus local tables can only be created from coordinator for now */
+	EnsureCoordinator();
+
+	EnsureTableOwner(relationOid);
+
+	/* we allow creating citus local tables only from relations with RELKIND_RELATION */
```

Comment:
I think we should be allowing any kind of relation

Reply:
Actually `EnsureRelationKindSupported` allows us to create the citus table if the table is 
* a regular table or
* a foreign table or
* a partitioned table and
* is not an inherited table.

This check is to prevent views, sequences etc.